src/bin/pgcopydb/pgcmd.c

```diff
@@ -350,8 +350,9 @@ pg_dump_db(PostgresPaths *pgPaths,
 
 	SafeURI safeURI = { 0 };
 	bool pgpassword_found_in_env = env_exists("PGPASSWORD");
-	char PGPASSWORD[MAXCONNINFO] = { 0 };
-
+	size_t len = strlen(pguri) + 1;
+	char *PGPASSWORD = (char *) calloc(len, sizeof(char));
+	safeURI.password = (char *) calloc(len, sizeof(char));
```

Comment:
In the function `extract_connection_string_password` the safeURI.password memory allocation is dealt with via calling `strdup` again, so that's another wasted memory area, we can just remove that line entirely.